lib/internal/streams/pipeline.js

```diff
@@ -147,6 +150,28 @@ async function pump(iterable, writable, finish, { end }) {
   }
 }
 
+async function pumpToWeb(readable, writable, finish, { end }) {
+
+  if (isTransformStream(writable)) {
+    writable = writable.writable;
+  }
+
+  try {
+    const writer = writable.getWriter();
+    for await (const chunk of readable) {
+      await writer.write(chunk);
+    }
+
+    if (end) {
+      await writer.close();
+    }
+
+    finish();
+  } catch (err) {
+    finish(err);
```

Comment:
@ronag does this guarantee that all the WHATWG streams are closed correctly?

Reply:
No, we are missing a `await writer.abort()` here.